modules/apigee-organization/main.tf

```diff
@@ -50,6 +52,15 @@ resource "google_apigee_environment" "apigee_env" {
   deployment_type = each.value.deployment_type
   name            = each.key
   org_id          = google_apigee_organization.apigee_org.id
+
+  dynamic "node_config" {
+    for_each = var.billing_type == "PAYG" ? toset([1]) : toset([])
```

Comment:
i think the toset() is not needed. We typically use `condition ? []: [1]` ir `condition ? {} : {1=1}`